sources/PTYTextView.h

```diff
@@ -317,6 +322,9 @@ typedef enum {
 // Updates the preferences for semantic history.
 - (void)setSemanticHistoryPrefs:(NSDictionary *)prefs;
 
+// Is the main window active?
+- (void)setIsBackground:(BOOL)isBackground;
```

Comment:
I'm trying to break direct dependencies between `PseudoTerminal` and `PTYTextView`. Rather than have it call `setIsBackground:` directly from `windowDidBecomeKey:` and `windowDidResignKey:`, we can use notifications to keep this decoupled.

You can register for notifications like this, in -[PTYTextView initWithFrame:colorMap:]:

      [[NSNotificationCenter defaultCenter] addObserver:self
                                               selector:@selector(windowDidBecomeKey:)
                                                   name:NSWindowDidBecomeKeyNotification
                                                 object:nil];

      [[NSNotificationCenter defaultCenter] addObserver:self
                                               selector:@selector(windowDidResignKey:)
                                                   name:NSWindowDidResignKeyNotification
                                                 object:nil];

And add methods like these:

      - (void)windowDidBecomeKey:(NSNotification *)notification {
          if ([notification object] == [self window]) {
              [self setIsBackground:NO];
          }
      }

      - (void)windowDidResignKey:(NSNotification *)notification {
          if ([notification object] == [self window]) {
              [self setIsBackground:YES];
          }
      }

And then -setIsBackground can be a private method.